ArcGISRuntimeSDKQt_CppSamples/Layers/ListKmlContents/ListKmlContents.cpp

```diff
@@ -210,6 +215,95 @@ void ListKmlContents::processSelectedNode(const QString& nodeName)
   }
 }
 
+void ListKmlContents::getViewpointFromKmlViewpoint(KmlNode* node)
+{
+  const KmlViewpoint kmlViewpoint = node->viewpoint();
+
+  if (!kmlViewpoint.isEmpty())
+  {
+    // altitude adjustment is needed for all but Absolute altitude mode
+    m_needsAltitudeFixed = (kmlViewpoint.altitudeMode() != KmlAltitudeMode::Absolute);
+
+    switch (kmlViewpoint.type())
+    {
+    case KmlViewpointType::LookAt:
+      m_viewpoint = Viewpoint(kmlViewpoint.location(), Camera(kmlViewpoint.location(), kmlViewpoint.range(),
```

Comment:
I guess you could shorten these 2 case statements if you had create the camera before hand... looks the same in both scenarios?

Reply:
The two `Camera` objects are different, as the first has a distance parameter specified and the second doesn't